libraries/lib-sentry-reporting/SentryReport.cpp

```diff
@@ -0,0 +1,430 @@
+/*!********************************************************************
+
+ Audacity: A Digital Audio Editor
+
+ @file SentryReport.cpp
+ @brief Define a class to report errors to Sentry.
+
+ Dmitry Vedenko
+ **********************************************************************/
+
+#include "SentryReport.h"
+
+#include <chrono>
+#include <cstring>
+#include <mutex>
+
+#include <algorithm>
+#include <cctype>
+
+#include <rapidjson/document.h>
+#include <rapidjson/writer.h>
+#include <rapidjson/prettywriter.h>
+
+#include <wx/platinfo.h>
+#include <wx/log.h>
+
+#include "CodeConversions.h"
+#include "Uuid.h"
+
+#include "IResponse.h"
+#include "NetworkManager.h"
+
+#include "SentryRequestBuilder.h"
+
+namespace audacity
+{
+namespace sentry
+{
+namespace
+{
+//! This small class is a thread safe store for the information 
+//! we want to add to the exception before the exception occurs.
+//! For example, we may log SQLite3 return codes here, as otherwise 
+//! they wont be available when everything fails
```

Comment:
The comment above are organized by Doxygen into one detailed comment on the class, with no brief comment

I usually write

`
//! A brief description
/*!
And longer comments
*/
`